pkg/scalers/aws_sigv4.go

```diff
@@ -0,0 +1,111 @@
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/aws/aws-sdk-go-v2/service/amp"
+	awsutils "github.com/kedacore/keda/v2/pkg/scalers/aws"
+)
+
+// SigV4Config configures signing requests with SigV4.
+type SigV4Config struct {
+	Enabled bool   `yaml:"enabled,omitempty"`
+	Region  string `yaml:"region,omitempty"`
+}
+
+type awsConfigMetadata struct {
+	awsRegion        string
+	awsAuthorization awsutils.AuthorizationMetadata
+}
+
+// Custom round tripper to sign requests
+type roundTripper struct {
+	client *amp.Client
+	region string
+}
+
+var (
+	// ErrAwsAMPNoAwsRegion is returned when "awsRegion" is missing from the config.
+	ErrAwsAMPNoAwsRegion = errors.New("no awsRegion given")
+)
+
+func (rt *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+
+	cred, err := rt.client.Options().Credentials.Retrieve(req.Context())
+	if err != nil {
+		return nil, err
+	}
+	// Sign request
+	hasher := sha256.New()
+	reqCxt := v4.SetPayloadHash(req.Context(), hex.EncodeToString(hasher.Sum(nil)))
+	reqHash := v4.GetPayloadHash(reqCxt)
+	err = rt.client.Options().HTTPSignerV4.SignHTTP(req.Context(), cred, req, reqHash, "aps", rt.region, time.Now())
+	if err != nil {
+		return nil, err
+	}
+	// Create default transport
+	transport := &http.Transport{}
+
+	// Send signed request
+	return transport.RoundTrip(req)
+}
+
+func parseAwsAMPMetadata(config *ScalerConfig) (*awsConfigMetadata, error) {
+	meta := awsConfigMetadata{}
+
+	if val, ok := config.TriggerMetadata["awsRegion"]; ok && val != "" {
+		meta.awsRegion = val
+	} else {
+		return nil, ErrAwsAMPNoAwsRegion
+	}
+
+	auth, err := awsutils.GetAwsAuthorization(config.TriggerUniqueKey, config.PodIdentity, config.TriggerMetadata, config.AuthParams, config.ResolvedEnv)
+	if err != nil {
+		return nil, err
+	}
+
+	meta.awsAuthorization = auth
+	return &meta, nil
+}
+
+// NewSigV4RoundTripper returns a new http.RoundTripper that will sign requests
+// using Amazon's Signature Verification V4 signing procedure. The request will
+// then be handed off to the next RoundTripper provided by next. If next is nil,
+// http.DefaultTransport will be used.
+//
+// Credentials for signing are retrieving used the default AWS credential chain.
+// If credentials could not be found, an error will be returned.
+func NewSigV4RoundTripper(config *ScalerConfig) (http.RoundTripper, error) {
+	metadata, err := parseAwsAMPMetadata(config)
+	if err != nil {
+		return nil, err
+	}
+	awsCfg, err := awsutils.GetAwsConfig(context.Background(), metadata.awsRegion, metadata.awsAuthorization)
+	if err != nil {
+		return nil, err
+	}
+
+	triggerMetadata := config.TriggerMetadata
+	if triggerMetadata == nil {
+		return nil, fmt.Errorf("trigger metadata cannot be nil")
+	}
+
+	awsRegion := triggerMetadata["awsRegion"]
+	if awsRegion == "" {
+		return nil, fmt.Errorf("awsRegion not configured in trigger metadata")
+	}
```

Comment:
Do we need this? I mean, `metadata` already has `awsRegion` but also if you don't set it, the underlying client uses default region